src/posts/2020-12-21-ebpf-updates/index.md

```diff
@@ -0,0 +1,393 @@
+  you can directly build your software. Of the two available templates, the
+  simplest one (`minimal`) manipulates an eBPF program that simply logs the PID
+  of the process that calls it. The more advanced template (`bootstrap`) sets
+  up an application with more advanced features like eBPF maps, read-only
+  configuration variables, eBPF ring buffer, or CO-RE which needs a BTF (BPF
+  Type Format) description of the kernel's internals. This means that using
+  these features gets simple and immediate, all is set up for you in the
+  template. This article goes into a thorough description of the mechanisms
+  involved. This is a long read, but well worth it if you want to program
+  applications working with eBPF.
+
+* [_A first xBGP plugin_](https://pluginized-protocols.org/xbgp/2020/11/29/xbgp-hello.html),
+  from Thomas Wirtgen.  
+  As a follow-up from the link to the paper for xBGP in the previous issue of
+  these _eBPF Updates_, this is the introduction of a first eBPF-based xBGP
+  plugin. The idea is that, quoting the post, “_a network operator would like
+  to ignore the BGP UPDATE messages that contain an unknown attribute. A
+  practical example of this usage is when problems with the processing of BGP
+  Path attribute 128 caused the failure of BGP sessions_”. All steps required
+  for running this example are provided. The code is hosted on
+  [GitHub](https://github.com/pluginized-protocols/xbgp_plugins.git), but there
+  is also a
+  [Dockerfile](https://github.com/pluginized-protocols/libxbgp/blob/master/misc/Dockerfile_xbgp)
+  packaging all the required elements.
+
+* [_Integrating an eBPF-based firewall into the TezEdge node with multipass validations_](https://medium.com/simplestaking/integrating-an-ebpf-based-firewall-into-the-tezedge-node-with-multipass-validations-769d4c6ccd93),
+  from Juraj Selep.  
+  TezEdge peer-to-peer nodes validate blocks for the decentralized
+  [Tezos](https://en.wikipedia.org/wiki/Tezos) blockchain, providing smart
+  contracts. Blockchain networks are subject to DDoS (Distributed Denial of
+  Service) attacks, generally mitigated with a firewall. In the current case,
+  XDP is used to implement it. The eBPF program checks that each connection
+  starts with a valid and unique proof of work, making it expensive for an
+  adversary to start many connections. This is further integrated with the
+  “multipass validation” scheme that aims at detecting erroneous blocks as soon
+  as possible. Note that the eBPF programs are written in Rust.
+
+* [_Snooping MySQL passwords with eBPF_](https://distributeddba.com/blog/database-administration/data-security/snooping-mysql-passwords-with-ebpf/),
+  from Derek Downey.  
+  Providing unencrypted password through command prompt or environment
+  variables may present a security issue. This short post relies on bcc's
+  execsnoop tool to show that MySQL passwords can be retrieved by examining the
+  arguments passed to the system call, on the kernel side.
```

Comment:
Not sure we should include that one. It's demonstrating why passing passwords in clear text to MySQL is insecure with bcc's execsnoop. I don't think that makes sense given execsnoop requires root privileges. Using the interaction prompt to provide the password to MySQL is not going to protect against a privileged user either.

More generally, I'm afraid it could give the wrong impression that BPF tracing tools are dangerous and can be used for privilege escalation.

Reply:
I was a bit hesitant as well. Ok I'll remove it.